lib/src/public/interfaces/stream_bloc_states.dart

```diff
@@ -0,0 +1,28 @@
+abstract class StreamBlocStates<Event extends Object?, State extends Object?> {
```

Comment:
Looks useful and I'm willing to accept those changes, but I would prefer keeping only two methods – `whereState` and `whereUnique` with optional unnamed predicate arguments.